src/create-or-update-comment.ts

```diff
@@ -118,18 +118,25 @@ function appendSeparatorTo(body: string, separator: string): string {
   }
 }
 
+function truncateBody(
+  body: string
+) {
+  // 65536 characters is the maximum allowed for issue comments.
+  if (body.length > 65536) {
+    core.warning(`Comment body is too long. Truncating to 65536 characters.`)
+    return body.substring(0, 65536)
+  }
+  return body
+}
+
 async function createComment(
   octokit,
   owner: string,
   repo: string,
   issueNumber: number,
   body: string
 ): Promise<number> {
-  // 65536 characters is the maximum allowed for issue comments.
-  if (body.length > 65536) {
-    core.warning(`Comment body is too long. Truncating to 65536 characters.`)
-    body = body.substring(0, 65536)
-  }
```

Comment:
How about we just move this block of code to [here](https://github.com/peter-evans/create-or-update-comment/blob/main/src/create-or-update-comment.ts#L236). Then it should handle both create/update cases.